storage/csv/datasource.go

```diff
@@ -112,13 +139,163 @@ func (rs *RecordStream) Close() error {
 	return nil
 }
 
-func parseDataTypes(row []string) []octosql.Value {
-	resultRow := make([]octosql.Value, len(row))
-	for i, v := range row {
-		resultRow[i] = execution.ParseType(v)
+var offsetPrefix = []byte("csv_offset")
+
+func (rs *RecordStream) RunWorker(ctx context.Context) {
+	for { // outer for is loading offset value and moving file iterator
+		tx := rs.stateStorage.BeginTransaction().WithPrefix(rs.streamID.AsPrefix())
+
+		err := rs.loadOffset(tx)
+		if err != nil {
+			log.Fatalf("csv worker: couldn't reinitialize offset for csv read batch worker: %s", err)
+			return
+		}
+
+		tx.Abort() // We only read data above, no need to risk failing now.
+
+		// Moving file iterator by `rs.offset`
+		for i := 0; i < rs.offset; i++ {
+			_, err := rs.readRecordFromFileWithInitialize()
```

Comment:
Can't this be done more efficiently? I mean move the offset all at once?
Like initialize first and then just move by offset lines?

Reply:
I'm afraid not with a use of those packages, the way to handle this better (possibly) is to store bytes read after every line and then use Seek on the file itself but it's pretty disgusting imho :(

Reply:
Yea, don't optimize it for now.